### PR TITLE
epics added

### PR DESCRIPTION
**Pull request by samioan**
#42 Check option now advances to the next phase.
#41 Pot sharing fixed.
#37 Log reducer refactored, log now contains more detailed messages about every action.
#36 Epics created, extract business from game reducer.
#29 Game phases are distinct. Alerts removed and replaced with log messages.

Hope everything's okay! Get well soon.

Co-authored-by: Stathis, Konstantinos <StathisK@kotsovolos.gr>

### CODE REVIEW

File: src/models/game/actions.js [hunk 1]
```diff
@@ -8,8 +8,19 @@ const Action = (type) => {
 };
 
 export const check = Action("CHECK");
+export const gameChecked = Action("GAME_CHECKED");
+export const gameWin = Action("GAME_WIN");
+export const gameLose = Action("GAME_LOSE");
+export const gameTie = Action("GAME_TIE");
```
Comment (stacon) [inline src/models/game/actions.js]: GAME prefix seems irrelevant with the actual actions. Check also for game folded. They are more player related, aren't they?

File: src/models/game/epics.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+          aiMoney: newAiMoney - bet,
+          pot: newPot + bet * 2,
+          playerBet: newPlayerBet + bet,
+          aiBet: newAiBet + bet,
+        });
+      } else if (newPlayerMoney <= 0) {
+        return betNotRaised({});
+      }
+    })
+  );
```
Comment (stacon) [inline src/models/game/epics.js]: you dont need bet not raused. You better use the condintion with a filter operator before map

File: src/models/game/epics.js [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        );
+        newDeck.splice(0, 1);
+        return cardReplaced({
+          deck: newDeck,
+          playerHand: newPlayerHand,
+          changedPlayerHand: newChangedPlayerHand,
+        });
+      } else return cardNotReplaced({});
+    })
+  );
```
Comment (stacon) [inline src/models/game/epics.js]: Same as above

File: src/models/game/epics.js [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+    map(({ payload }) => {
+      const newPlayerMoney = playerMoney(state$.value);
+      const newAiMoney = aiMoney(state$.value);
+      const newPlayerBet = playerBet(state$.value);
+      const newAiBet = aiBet(state$.value);
+      const newPot = pot(state$.value);
+      const newPhase = phase(state$.value);
+      const newPlayerHand = playerHand(state$.value).slice();
+      const newAiHand = aiHand(state$.value).slice();
+
```
Comment (stacon) [inline src/models/game/epics.js:164]: common properties from the object below can be created here under a const commonPayload and be spread in the objects below and replace only the different properties.

File: src/models/log/reducer.js [hunk 5]
```diff
@@ -1,63 +1,131 @@
...
     }
 
-    case replace.type: {
+    case cardReplaced.type: {
       const newLogger = state.logger.slice();
       newLogger.push("Player trades a card.");
-
+      newLogger.push(
+        "You now have " + handCheckToMsg(action.payload.playerHand) + "."
+      );
```
Comment (stacon) [inline src/models/log/reducer.js]: Try instead of slicing to use copy and array spread methods rg const newArray = [...previousArray, 'Another Message'].
Even better, create an action that receives a single string or array of strings and an epic should be responsible to append the logger messages

**Review by stacon — CHANGES_REQUESTED** [review]
Check comments

File: src/models/game/actions.js (hunk 1 above)
Comment (samioan) [reply src/models/game/actions.js]: Done.

File: src/models/game/epics.js (hunk 2 above)
Comment (samioan) [reply src/models/game/epics.js]: It is done.

File: src/models/game/epics.js (hunk 4 above)
Comment (samioan) [reply src/models/game/epics.js:164]: Done and done.

File: src/models/log/reducer.js (hunk 5 above)
Comment (samioan) [reply src/models/log/reducer.js]: It has been done.

File: src/models/game/epics.js (hunk 3 above)
Comment (samioan) [reply src/models/game/epics.js]: Done.

File: src/components/game/components/intro/components/forecast/Forecast.js [hunk 6]
```diff
@@ -1,60 +1,80 @@
...
-            <p className="text">
-              Grab some hot chocolate and play Poker because it's cold outside!
-            </p>
-          ) : null}
-        </div>
-      ) : null}
+const Forecast = ({ name, temp, description, onClickForecastHandler }) => (
+  <div>
+    <div>
+      <Button id={"Weather Forecast"} onClick={onClickForecastHandler} />
```
Comment (stacon) [inline src/components/game/components/intro/components/forecast/Forecast.js]:
```suggestion
      <Button id="Weather Forecast" onClick={onClickForecastHandler} />
```

**Review by stacon — CHANGES_REQUESTED** [review]
Check the minor suggestion

File: src/components/game/components/intro/components/forecast/Forecast.js (hunk 6 above)
Comment (samioan) [reply src/components/game/components/intro/components/forecast/Forecast.js]: Ok.

**Review by stacon — APPROVED** [review]

**Review by stacon — APPROVED** [review]
